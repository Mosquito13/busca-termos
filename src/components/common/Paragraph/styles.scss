.paragraph {
  font-family: inherit;
  font-size: inherit;
  color: currentColor;

  &--bold {
    font-weight: 600;
  }

  &--small {
    font-size: 1.2rem;
  }
}
