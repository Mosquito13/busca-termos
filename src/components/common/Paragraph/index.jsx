import classNames from 'classnames';
import PropTypes from 'prop-types';

import './styles.scss';

const Paragraph = ({ bold, smallFont, children }) => {
  const classes = classNames(
    'paragraph',
    bold && 'paragraph--bold',
    smallFont && 'paragraph--small'
  );

  return <p className={classes}>{children}</p>;
};

Paragraph.propTypes = {
  /**
   * Indica se e negrito
   */
  bold: PropTypes.bool,
  /**
   * Indica se deve renderizar com uma fonte menor do que o normal
   */
  smallFont: PropTypes.bool
};

export default Paragraph;
