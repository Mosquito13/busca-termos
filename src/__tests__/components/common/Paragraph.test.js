import renderer from 'react-test-renderer';

import Paragraph from '../../../components/common/Paragraph';

describe(`${Paragraph.name}`, () => {
  describe('snapshots', () => {
    it('should render default', () => {
      const tree = renderer.create(<Paragraph value="text" />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with bold font', () => {
      const tree = renderer.create(<Paragraph bold value="text" />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with small font', () => {
      const tree = renderer
        .create(<Paragraph smallFont value="text" />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
