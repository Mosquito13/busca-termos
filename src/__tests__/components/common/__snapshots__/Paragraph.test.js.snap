// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph snapshots should render default 1`] = `
<p
  className="paragraph"
>
  text
</p>
`;

exports[`Paragraph snapshots should render with bold font 1`] = `
<p
  className="paragraph paragraph--bold"
>
  text
</p>
`;

exports[`Paragraph snapshots should render with small font 1`] = `
<p
  className="paragraph paragraph--small"
>
  text
</p>
`;
